### PR TITLE
wxGUI/vselect: fix show error message if selected map layer type is raster

### PR DESCRIPTION
**Pull request by tmszi**
**To Reproduce:**

1. Start wxGUI `g.gui`
2. Display some raster map (and select this map layer in the Layer Manager map tree if it isn't selected)
3. On the Map Display window from toolbar choose Select vector feature(s) tool
4. Left click on the aspect raster map to select feature

**Default behavior:**

You get error message which isn't understandable for user.

wxgui_vselect_error_message_def

**Expected behavior:**

Select vector feature(s) tool support vector map layer type only and you get understandable error message.

wxgui_vselect_error_message_exp

### CODE REVIEW

File: gui/wxpython/gui_core/vselect.py [hunk 1]
```diff
@@ -269,10 +269,21 @@ def GetSelectedMap(self):
             return None
 
         if layerSelected:
-            mapName = str(layerSelected)
-            if self.mapName is not None:
-                if self.mapName != mapName:
-                    self.Reset()
+            if layerSelected.type == 'raster':
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/vselect.py:272]: I think type != 'vector' would be better since there are also 3D rasters.

**Review by petrasovaa — CHANGES_REQUESTED** [review]

File: gui/wxpython/gui_core/vselect.py (hunk 1 above)
Comment (tmszi) [reply gui/wxpython/gui_core/vselect.py]: Yes you are right I have already fixed it.

File: gui/wxpython/gui_core/vselect.py [hunk 2]
```diff
@@ -269,10 +269,21 @@ def GetSelectedMap(self):
...
-                if self.mapName != mapName:
-                    self.Reset()
+            if layerSelected.type != 'vector':
+                mapName = None
+                self.UnregisterMapEvtHandler()
+                GError(
+                    _(
+                        "Selected raster map layer <{}> "
+                        "isn't supported. Please select vector map layer. "
+                        "Operation canceled.".format(layerSelected),
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/vselect.py:279]:
```suggestion
                        "No vector map layer selected. Operation canceled."
```

File: gui/wxpython/gui_core/vselect.py [hunk 3]
```diff
@@ -269,10 +269,19 @@ def GetSelectedMap(self):
...
-                if self.mapName != mapName:
-                    self.Reset()
+            if layerSelected.type != 'vector':
+                mapName = None
+                self.UnregisterMapEvtHandler()
+                GError(
+                    _(
+                        "No vector map layer selected. Operation canceled."
+                    ),
+                )
```
Comment (petrasovaa) [inline gui/wxpython/gui_core/vselect.py:279]:
```suggestion
                GError(_("No vector map layer selected. Operation canceled."))
```

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**